src/modules/map/store/GeolocationObserver.js

```diff
@@ -3,29 +3,19 @@ import { $injector } from '../../../injection';
 import { setPosition, setAccuracy, setDenied, setTracking } from './geolocation.action';
 import { changeCenter, setFit } from './position.action';
 import { addLayer, removeLayer } from './layers.action';
+import { BaObserver } from '../../BaObserver';
 
 /**
  * Id of the layer used for geolocation visualization
  */
 export const GEOLOCATION_LAYER_ID = 'geolocation_layer';
 
-export class GeolocationHandler {
-	constructor(store) {
-		const {
-			TranslationService: translationService,
-			CoordinateService: coordinateService,
-			EnvironmentService: environmentService,
-			MapService: mapService
-		}
-			= $injector.inject('TranslationService', 'CoordinateService', 'EnvironmentService', 'MapService');
-		this._translationService = translationService;
-		this._coordinateService = coordinateService;
-		this._environmentService = environmentService;
-		this._mapService = mapService;
```

Comment:
what is the reason to move the service-objects directly to they corresponding usage? Is there a potential performance issue to get every time a service from the injector?

Reply:
There aren't  any performance issues to be expected, both approaches use  "call-by-reference".
My goal here was just to separate the unneeded local state from the necessary one in order to minimize side effects.